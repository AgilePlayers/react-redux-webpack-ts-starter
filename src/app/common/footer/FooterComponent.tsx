import React from 'react';

import './footer.scss';

const FooterComponent: React.SFC = () => (
  <div className="footer">
    <p className="footer-paragraph">
      Created by
      <span className="footer-text-handwritten">Kamil Bysiec</span> from
    </p>
    <a
      href="https://agileplayers.com"
      target="_blank"
      rel="noopener noreferrer"
    >
      <img
        src="https://agileplayers.com/Static/dist/img/logo_full.svg"
        alt="agile players logo"
        className="footer-link-logo"
      />
    </a>
  </div>
);

export default FooterComponent;
