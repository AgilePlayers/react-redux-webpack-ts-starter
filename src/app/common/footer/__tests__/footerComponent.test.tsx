import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import FooterComponent from '../FooterComponent';

describe('common / footer / FooterComponent', () => {
  let component: ShallowWrapper;

  beforeEach(() => {
    component = shallow(<FooterComponent />);
  });

  test('should render', () => {
    expect(component.length).toEqual(1);
  });

  test('should match snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  test('should contains elements with "footer" and "footer-link-logo" class', () => {
    expect(component.find('div.footer').length).toEqual(1);
    expect(component.find('img.footer-link-logo').length).toEqual(1);
  });
});
