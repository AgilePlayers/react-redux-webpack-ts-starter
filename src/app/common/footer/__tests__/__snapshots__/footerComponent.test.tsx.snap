// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`common / footer / FooterComponent should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FooterComponent />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          className="footer-paragraph"
        >
          Created by
          <span
            className="footer-text-handwritten"
          >
            Kamil Bysiec
          </span>
           from
        </p>,
        <a
          href="https://agileplayers.com"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="agile players logo"
            className="footer-link-logo"
            src="https://agileplayers.com/Static/dist/img/logo_full.svg"
          />
        </a>,
      ],
      "className": "footer",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Created by",
            <span
              className="footer-text-handwritten"
            >
              Kamil Bysiec
            </span>,
            " from",
          ],
          "className": "footer-paragraph",
        },
        "ref": null,
        "rendered": Array [
          "Created by",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Kamil Bysiec",
              "className": "footer-text-handwritten",
            },
            "ref": null,
            "rendered": "Kamil Bysiec",
            "type": "span",
          },
          " from",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt="agile players logo"
            className="footer-link-logo"
            src="https://agileplayers.com/Static/dist/img/logo_full.svg"
          />,
          "href": "https://agileplayers.com",
          "rel": "noopener noreferrer",
          "target": "_blank",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "agile players logo",
            "className": "footer-link-logo",
            "src": "https://agileplayers.com/Static/dist/img/logo_full.svg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "a",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className="footer-paragraph"
          >
            Created by
            <span
              className="footer-text-handwritten"
            >
              Kamil Bysiec
            </span>
             from
          </p>,
          <a
            href="https://agileplayers.com"
            rel="noopener noreferrer"
            target="_blank"
          >
            <img
              alt="agile players logo"
              className="footer-link-logo"
              src="https://agileplayers.com/Static/dist/img/logo_full.svg"
            />
          </a>,
        ],
        "className": "footer",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Created by",
              <span
                className="footer-text-handwritten"
              >
                Kamil Bysiec
              </span>,
              " from",
            ],
            "className": "footer-paragraph",
          },
          "ref": null,
          "rendered": Array [
            "Created by",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Kamil Bysiec",
                "className": "footer-text-handwritten",
              },
              "ref": null,
              "rendered": "Kamil Bysiec",
              "type": "span",
            },
            " from",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="agile players logo"
              className="footer-link-logo"
              src="https://agileplayers.com/Static/dist/img/logo_full.svg"
            />,
            "href": "https://agileplayers.com",
            "rel": "noopener noreferrer",
            "target": "_blank",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "agile players logo",
              "className": "footer-link-logo",
              "src": "https://agileplayers.com/Static/dist/img/logo_full.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "a",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
<<<<<<< HEAD:src/app/common/footer/__tests__/__snapshots__/footerComponent.test.tsx.snap
          "getSnapshotBeforeUpdate": true,
=======
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
>>>>>>> npm-update:src-ts/app/common/footer/__tests__/__snapshots__/footerComponent.test.tsx.snap
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
